### PR TITLE
Bugfix for wrong default repository selection in reconciler

### PR DESCRIPTION
**Pull request by tobiscr**

### CODE REVIEW

File: cmd/mothership/local/cmd.go [hunk 1]
```diff
@@ -77,15 +77,15 @@ func RunLocal(o *Options) error {
 
 	l.Infof("Local installation started with kubeconfig %s", o.kubeconfigFile)
 
-	cluster, err := prepareClusterState(o)
+	clState, err := prepareClusterState(o)
```
Comment (tobiscr) [inline cmd/mothership/local/cmd.go:80]: Just making the linter happy

File: pkg/reconciler/chart/factory.go [hunk 2]
```diff
@@ -241,7 +243,11 @@ func (f *DefaultFactory) downloadArchive(URL, dstDir string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	defer resp.Body.Close()
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
```
Comment (tobiscr) [inline pkg/reconciler/chart/factory.go:247]: making the linter happy

File: pkg/reconciler/chart/factory_test.go [hunk 3]
```diff
@@ -347,10 +346,10 @@ func Test_ExternalGitComponent(t *testing.T) {
 func validateWorkspace(fileName, componentName, serverURL string) error {
 	fileURL := fmt.Sprintf("%s/%s", serverURL, fileName)
 	dirname := fmt.Sprintf("%x-%s", sha1.Sum([]byte(fileURL)), componentName) //nolint
-	filepath := path.Join(storageDir, dirname, componentName)
+	filePath := path.Join(storageDir, dirname, componentName)
```
Comment (tobiscr) [inline pkg/reconciler/chart/factory_test.go:349]: making the linter happy

File: pkg/reconciler/instances/istio/istio_test.go [hunk 4]
```diff
@@ -313,7 +313,7 @@ type TestCommanderResolver struct {
 	cmder istioctl.Commander
 }
 
-func (tcr TestCommanderResolver) GetCommander(version istioctl.Version) (istioctl.Commander, error) {
+func (tcr TestCommanderResolver) GetCommander(_ istioctl.Version) (istioctl.Commander, error) {
 	if tcr.err != nil {
```
Comment (tobiscr) [inline pkg/reconciler/instances/istio/istio_test.go:317]: making the linter happy

File: pkg/db/sqlite.go [hunk 5]
```diff
@@ -177,8 +177,5 @@ func (scf *sqliteConnectionFactory) resetFile() error {
 	if err != nil {
 		return err
 	}
-	if err := file.Close(); err != nil {
-		return err
-	}
-	return nil
+	return file.Close()
```
Comment (tobiscr) [inline pkg/db/sqlite.go:180]: fix linter issue

File: pkg/reconciler/instances/cleaner/pkg/cleanup/cli.go [hunk 6]
```diff
@@ -72,11 +72,8 @@ func (cmd *CliCleaner) Run() error {
 	if err := cmd.deleteKymaNamespaces(); err != nil {
 		return err
 	}
-	if err := cmd.waitForNamespaces(); err != nil {
-		return err
-	}
 
-	return nil
+	return cmd.waitForNamespaces()
```
Comment (tobiscr) [inline pkg/reconciler/instances/cleaner/pkg/cleanup/cli.go:76]: fix linter issue

File: pkg/reconciler/instances/cleaner/pkg/cleanup/cli.go [hunk 7]
```diff
@@ -201,11 +198,7 @@ func (cmd *CliCleaner) removeResourcesFinalizers() error {
 		return err
 	}
 
-	if err := cmd.removeCustomResourcesFinalizers(); err != nil {
-		return err
-	}
-
-	return nil
+	return cmd.removeCustomResourcesFinalizers()
```
Comment (tobiscr) [inline pkg/reconciler/instances/cleaner/pkg/cleanup/cli.go:201]: fix linter issue

File: pkg/reconciler/chart/factory.go [hunk 8]
```diff
@@ -53,11 +53,13 @@ type DefaultFactory struct {
...
+func NewFactory(storageDir string, logger *zap.SugaredLogger) (*DefaultFactory, error) {
 	factory := &DefaultFactory{
-		storageDir:     storageDir,
-		logger:         logger,
-		kymaRepository: repo,
+		storageDir: storageDir,
+		logger:     logger,
+		kymaRepository: &reconciler.Repository{
+			URL: defaultRepositoryURL,
+		},
```
Comment (tobiscr) [inline pkg/reconciler/chart/factory.go:62]: The bugfix ;)

**Review by jeremyharisch — APPROVED** [review]

**Comment by tobiscr** [comment]
Closing caused by breaking CLI change

**Pull request closed without merging**